### ISSUE TITLE
Assigning parameter to tensor doesn't pass gradient info

### ISSUE THREAD

**Issue by InfProbSciX** [pytorch/pytorch#77565]
## Issue description

In the example below, assigning a parameter to tensor doesn't seem to pass gradient information.

## Code example

```python
import torch
torch.__version__  # '1.11.0'
z = torch.zeros(3, 2)

z[:, 0] = torch.nn.Parameter(torch.tensor(1.0))
z

# tensor([[1., 0.],
#         [1., 0.],
#         [1., 0.]])  # <------------

z[:, [0]] = torch.nn.Parameter(torch.tensor(1.0))

# tensor([[1., 0.],
#         [1., 0.],
#         [1., 0.]], grad_fn=<CopySlices>)  # <------------
```

... is this behaviour expected?

**Comment by ezyang** [comment]
No, non-advanced indexing should propagate gradients, looks like a bug.

**Comment by albanD** [comment]
Note that it doesn't require to use `Parameter`, the same thing happens with plain Tensors that require gradients.
Also this only happens if we use a scalar Tensor, so I guess some convertion to plain python Number in the advanced indexing code?

Note that as a workaround, you can use `torch.nn.Parameter(torch.tensor((1.0,)))` that is a 1D Tensor with a single value.
